packages/dds/tree/src/simple-tree/types.ts

```diff
@@ -38,6 +38,69 @@ import { tryGetSchema } from "./treeNodeApi.js";
  */
 export type Unhydrated<T> = T;
 
+export interface NodeChangedEventArgs {
+	readonly changedProperties: ReadonlySet<string>;
+}
+
+export interface TreeChangeEventsBase {
+	/**
+	 * Emitted by a node after a batch of changes has been applied to the tree, when something changed anywhere in the
+	 * subtree rooted at it.
+	 *
+	 * @remarks
+	 * This event is not emitted when the node itself is moved to a different location in the tree or removed from the tree.
+	 * In that case it is emitted on the _parent_ node, not the node itself.
+	 *
+	 * The node itself is part of the subtree, so this event will be emitted even if the only changes are to the properties
+	 * of the node itself.
+	 *
+	 * For remote edits, this event is not guaranteed to occur in the same order or quantity that it did in
+	 * the client that made the original edit.
+	 *
+	 * When it is emitted, the tree is guaranteed to be in-schema.
+	 */
+	treeChanged(): void;
+}
+
+export interface TreeChangeEventsInternal extends TreeChangeEventsBase {
```

Comment:
`TreeChangeEventsInternal` adds nodeChanged with the payload, so the code that _emits_ the events can emit them with the payload. That code is agnostic to the node kind so splitting the interface for object/array/map didn't seem to make sense in this case.